internal/jimm/cloudcredential.go

```diff
@@ -350,3 +350,38 @@ func (j *JIMM) getCloudCredentialAttributes(ctx context.Context, cred *dbmodel.C
 	}
 	return attr, nil
 }
+
+// CopyCredential copies a cloud credential from one user to another.
+func (j *JIMM) CopyCredential(ctx context.Context, originalUser *openfga.User, newUser *openfga.User, cred names.CloudCredentialTag) (names.CloudCredentialTag, []jujuparams.UpdateCredentialModelResult, error) {
+	op := errors.Op("jimm.CopyCredential")
+
+	credential, err := j.GetCloudCredential(ctx, originalUser, cred)
```

Comment:
where do we check authorization.. is the original user allowed to set cloud credentials for the new user -> is the original user administrator of the "new user" service account?

Reply:
Perhaps it's a bit too inconsistent with some of the other business logic methods, but for service accounts the auth checks are all done at the API layer in `jujuapi` and then that decides whether to call the copyCredential method etc.

Reply:
perhaps we should move the authz check to make it consistent?

Reply:
Made a card for it https://warthogs.atlassian.net/browse/JUJU-7389